awx/main/models/notifications.py

```diff
@@ -78,6 +124,25 @@ def notification_class(self):
     def save(self, *args, **kwargs):
         new_instance = not bool(self.pk)
         update_fields = kwargs.get('update_fields', [])
+
+        # preserve existing notification messages if not overwritten by new messages
+        if not new_instance:
```

Comment:
I must be missing something. I don't see any variables modified except for the ones created locally in this if condition.